### ISSUE TITLE
utf8proc 1.2 release

### ISSUE THREAD

**Issue by stevengj** [JuliaLang/utf8proc#30]
Now that we have nicer documentation (thanks to #29), as far as I can tell we are ready for a 1.2 release as soon as the version number is updated.

(I don't think it's necessary to wait until we can get the HTML docs into the release tarball; there are equivalent docs in `utf8proc.h`, after all, and in any case the lack of formatted docs in the tarball is not a regression.)

@jiahao, @tkelman, @nalimilan, @StefanKarpinski, your thoughts?

**Comment by tkelman** [comment]
Fine by me. Worth doing a sanity check (maybe you have already) that things are still fine when current utf8proc master is linked into Julia?

**Comment by stevengj** [comment]
I just tried it, and all the Julia tests pass on my machine with the utf8proc master.

**Comment by jakebolewski** [comment]
+1 it would be great to fix the unicode character width issues

**Comment by stevengj** [comment]
Hmm, one final tweak: looking at this, it seems like we need `UTF8PROC_VERSION_MAJOR` etc. `#define`s in `utf8proc.h` so that we can check the API version at compile-time.  This will allow callers such as Julia to use e.g. `utf8proc_charwidth` for utf8proc-1.2 or later, and fall back to `wcwidth` for systems with older utf8proc versions.

**Comment by tkelman** [comment]
Is calling runtime `utf8proc_version` not sufficient?

**Comment by nalimilan** [comment]
`#define`s can always be useful.

I'd just like to add support for `make install DESTDIR` before the new release. Patch coming.

**Comment by stevengj** [comment]
Calling the runtime version is not sufficient if you want to decide at *compile-time* whether you can call a function like `utf8proc_charwidth` or fall back to `wcwidth`.  For example:
```
#if UTF8PROC_VERSION_MAJOR > 1 || (UTF8PROC_VERSION_MAJOR == 1 && UTF8PROC_VERSION_MINOR >= 2)
...use utf8proc_charwidth...
#else
...use wcwidth...
#endif
```

**Comment by stevengj** [comment]
@nalimilan, it would be more standard to do `make prefix=DESTDIR install`

**Comment by tkelman** [comment]
> the runtime version is not sufficient if you want to decide at compile-time whether you can call a function like `utf8proc_charwidth` or fall back to `wcwidth`.

Yeah, good point.

> it is more standard to do `make prefix=DESTDIR install`

Distribution buildbots all use `DESTDIR`. It's "strongly recommended" practice - http://www.gnu.org/prep/standards/html_node/DESTDIR.html

**Comment by stevengj** [comment]
Thanks, I wasn't familiar with staged installs.  (I normally use automake and get all these things for free, but automake seems like overkill here).

**Comment by nalimilan** [comment]
OK, see https://github.com/JuliaLang/utf8proc/pull/31

**Comment by stevengj** [comment]
Anything else to check before we are good to go?

**Comment by nalimilan** [comment]
Fine with me.

**Comment by tkelman** [comment]
Looks like you haven't tagged yet, I realized that the Julia build on Windows does use the makefile here. We probably don't want to do the soname symlinking there either.

**Comment by tkelman** [comment]
Just checked deps/Makefile again - it doesn't use `make install`, it does `make libmojibake.a` and copies the static library and header into Julia's `usr`. So probably no big deal.

**Comment by stevengj** [comment]
Yes, the Julia build only makes the `.a` library, because it ends up getting linked into `libjulia` (by default).

**Comment by stevengj** [comment]
Okay, release tag is pushed and website is updated.

**Comment by stevengj** [comment]
(Unfortunately, I just noticed that there were some stray mojibake references remaining in the `bench/` subdirectory.  Not a critical issue since `bench/bench.c` is mainly for use by utf8proc developers, who should be using git anyway, but probably calls for a 1.2.1 release at some point.  Let's wait a couple weeks, though—at least until JuliaLang/julia#10659 is merged and gives utf8proc 1.2 wider testing, to see if any other problems turn up).
